### ISSUE TITLE
W3C references not fetching

### ISSUE THREAD

**Issue by ronaldtse** [relaton/relaton-w3c#8]
From @manuel489  in https://github.com/metanorma/mn-samples-ogc/pull/107
> But the W3C references seem not to fetch anything.

e.g.
```
* [[[html5,1]]], W3C: *HTML5*, W3C Recommendation, 2019 http://www.w3.org/TR/html5/
```

**Comment by andrew2net** [comment]
I think the reference should be:
```
 * [[[W3C HTML5,1]]], W3C: *HTML5*, W3C Recommendation, 2019 http://www.w3.org/TR/html5/
```
the relaton needs a prefix W3C to recognize a document type.

**Comment by manuel489** [comment]
@andrew2net 

In fact, the format that I'm using is:

````
* [[[html5,W3C html5]]], W3C: *HTML5*, W3C Recommendation, 2019 http://www.w3.org/TR/html5/
````

I also tested it on this way:

````
* [[[html5,W3C HTML5]]], W3C: *HTML5*, W3C Recommendation, 2019 http://www.w3.org/TR/html5/
````

(Which I think is what you meant to say.) 
And I get an error message:

````
...
[relaton-ietf] ("RFC 7946") fetching...
[relaton-ietf] ("RFC 7946") found RFC 7946
[relaton-w3c] ("W3C HTML5") fetching...
C:/tools/ruby25/lib/ruby/gems/2.5.0/gems/relaton-w3c-1.3.0/lib/relaton_w3c/scrapper.rb:43:in `fetch_title': undefined method `text' for nil:NilClass (NoMethodError)
        from C:/tools/ruby25/lib/ruby/gems/2.5.0/gems/relaton-w3c-1.3.0/lib/relaton_w3c/scrapper.rb:24:in `parse_page'
        from C:/tools/ruby25/lib/ruby/gems/2.5.0/gems/relaton-w3c-1.3.0/lib/relaton_w3c/hit.rb:12:in `fetch'
        from C:/tools/ruby25/lib/ruby/gems/2.5.0/gems/relaton-w3c-1.3.0/lib/relaton_w3c/w3c_bibliography.rb:29:in `get'
...
````

**Comment by andrew2net** [comment]
@manuel489 fixed the error in the version 1.3.1

**Comment by ronaldtse** [comment]
Thanks, since it's fixed, @manuel489 can you help run `bundle update` and retry?

**Comment by manuel489** [comment]
Ok, I have tested these following references:

````
[bibliography]
== References
* [[[html5,W3C HTML5]]]
* [[[dwbp,W3C dwbp]]]
* [[[dcat,W3C vocab-dcat]]]
````

and got this message,

````
[relaton-w3c] ("W3C HTML5") fetching...
[relaton-w3c] ("W3C HTML5") found HTML5
Bibliography: ERROR: No document identifier retrieved for W3C HTML5
[relaton-w3c] ("W3C dwbp") fetching...
[relaton-w3c] ("W3C vocab-dcat") fetching...
Crossreferences: (XML Line 000342): iso19101 does not have a corresponding anchor ID in the bibliography!
...
````

and this output:

![W3C](https://user-images.githubusercontent.com/33627611/91780181-ce91cb00-ebc4-11ea-9c5a-5e8048e448e1.PNG)


Even though an "error" message appeared for `W3C HTML5`, it did fetch something. Nonetheless, nothing is fetched for `W3C dwbp` and `W3C vocab-dcat`. Maybe I'm not using the correct identifier for those references. @andrew2net , what do you think?

**Comment by andrew2net** [comment]
> ```
> [relaton-w3c] ("W3C HTML5") fetching...
> [relaton-w3c] ("W3C HTML5") found HTML5
> Bibliography: ERROR: No document identifier retrieved for W3C HTML5
> ```

@manuel489 this error isn't from relaton. I think the metanorma expects a document identifier, but there aren't any document identifiers to scrape. Maybe we can make a document identifier from a reference. What do you think @ronaldtse?

**Comment by andrew2net** [comment]
> ```
> [relaton-w3c] ("W3C dwbp") fetching...
> [relaton-w3c] ("W3C vocab-dcat") fetching...
> ```

these references should be `"W3C Data on the Web Best Practices"` and `"W3C Data Catalog Vocabulary (DCAT)"` respectively.

> Crossreferences: (XML Line 000342): iso19101 does not have a corresponding anchor ID in the bibliography!

This message isn't from the relaton. I don't know what does it mean. Ping @ronaldtse @opoudjis.

**Comment by ronaldtse** [comment]
> @manuel489 this error isn't from relaton. I think the metanorma expects a document identifier, but there aren't any document identifiers to scrape. Maybe we can make a document identifier from a reference. What do you think?

@andrew2net yes I think we need to provide a document identifier. Let's just call it something like `W3C {code-name}`.

>>Crossreferences: (XML Line 000342): iso19101 does not have a corresponding anchor ID in the bibliography!
> This message isn't from the relaton. I don't know what does it mean.

I think it means that there is no anchor ID set in the bibliography item, i.e. `* [[[anchorID,name]]]`, `anchorID` is missing.

**Comment by andrew2net** [comment]
> Even though an "error" message seems to appear for `W3C HTML5`, it did fetch something. Nonetheless, nothing is fetched for `W3C dwbp` and `W3C vocab-dcat`. Maybe I'm not using the correct identifier for those references. @andrew2net , what do you think?

These errors are raised after the relaton fetch a document, by the other gem.
Looks like in the references you use the form of a shortcut of documents title, but the relaton-w3c search documents in a YAML file with only full titles. I.e. dwbp is a shortcut of a "Data on the Web Best Practices" title. The shortcut only appears in the document page link https://www.w3.org/TR/2017/REC-dwbp-20170131/. So we could implement searching by a shortcut.
@ronaldtse what do you think? Do we need it?

**Comment by andrew2net** [comment]
> @andrew2net yes I think we need to provide a document identifier. Let's just call it something like `W3C {code-name}`

@manuel489 what is `{code-name}`? Could you give an example?

**Comment by ronaldtse** [comment]
`{code-name}` is like `REC-dwbp-20170131` or `REC dwbp`.

**Comment by ronaldtse** [comment]
> Looks like in the references you use the form of a shortcut of documents title, but the relaton-w3c search documents in a YAML file with only full titles. I.e. dwbp is a shortcut of a "Data on the Web Best Practices" title. The shortcut only appears in the document page link https://www.w3.org/TR/2017/REC-dwbp-20170131/. So we could implement searching by a shortcut.

Yes we should allow searching by shortcut (let's call this the `docname` or `docid`).

**Comment by manuel489** [comment]
> Crossreferences: (XML Line 000342): iso19101 does not have a corresponding anchor ID in the bibliography!
> This message isn't from the relaton. I don't know what does it mean.

Probably this is not important because, when I performed the tests, I cut out half of the document content (including the bibliography, which is the last section) in order to compile faster.

**Comment by andrew2net** [comment]
implemented short reference and parsing docidentifier in v 1.3.2

**Comment by manuel489** [comment]
@andrew2net, I tested other W3C references:

````
[bibliography]
== References

* [[[html5,W3C HTML5]]]

* [[[dwbp,W3C dwbp]]]

* [[[dcat,W3C vocab-dcat]]]

* [[[w3cxml-0,W3C REC-xml]]], Extensible Markup Language (XML) 1.0 (Fifth Edition), W3C Recommendation, 26 November 2008, http://www.w3.org/TR/REC-xml

* [[[w3cxml-1,W3C xmlschema-1]]], XML Schema Part 1: Structures, http://www.w3.org/TR/xmlschema-1

* [[[w3cxml-2,W3C xmlschema-2]]], XML Schema Part 2: Datatypes, http://www.w3.org/TR/xmlschema-2

* [[[w3c,W3C REC-xml-names-19990114]]], Namespaces in XML, http://www.w3.org/TR/1999/REC-xml-names-19990114
````
and got this:
````
[relaton-w3c] ("W3C HTML5") fetching...
[relaton-w3c] ("W3C HTML5") found HTML5
[relaton-w3c] ("W3C dwbp") fetching...
[relaton-w3c] ("W3C dwbp") found Data on the Web Best Practices
[relaton-w3c] ("W3C vocab-dcat") fetching...
[relaton-w3c] ("W3C vocab-dcat") found Data Catalog Vocabulary (DCAT)
[relaton-w3c] ("W3C REC-xml") fetching...
[relaton-w3c] ("W3C REC-xml") found XML Signature Syntax and Processing Version 2.0
[relaton-w3c] ("W3C xmlschema-1") fetching...
C:/tools/ruby25/lib/ruby/gems/2.5.0/gems/relaton-w3c-1.3.2/lib/relaton_w3c/w3c_bibliography.rb:30:in `get': undefined method `title' for nil:NilClass (NoMethodError)
        from C:/tools/ruby25/lib/ruby/gems/2.5.0/gems/relaton-w3c-1.3.2/lib/relaton_w3c/processor.rb:19:in `get'
        from C:/tools/ruby25/lib/ruby/gems/2.5.0/gems/relaton-1.3.0/lib/relaton/db.rb:180:in `new_bib_entry'
...
````

`W3C xmlschema-1`, `W3C xmlschema-2`, and `W3C REC-xml-names-19990114` are not working. I tried several IDs..

**Comment by manuel489** [comment]
This one is not working either:
````
* [[[W3C,W3C css-color-3]]], W3C: CSS3 Color, 2018. https://www.w3.org/TR/css-color-3/[https://www.w3.org/TR/css-color-3/]
````

Message:
````
...
[relaton-w3c] ("W3C css-color-3") fetching...
C:/tools/ruby25/lib/ruby/gems/2.5.0/gems/relaton-w3c-1.3.2/lib/relaton_w3c/w3c_bibliography.rb:30:in `get': undefined method `title' for nil:NilClass (NoMethodError)
        from C:/tools/ruby25/lib/ruby/gems/2.5.0/gems/relaton-w3c-1.3.2/lib/relaton_w3c/processor.rb:19:in `get'
        from C:/tools/ruby25/lib/ruby/gems/2.5.0/gems/relaton-1.3.0/lib/relaton/db.rb:180:in `new_bib_entry'
        from C:/tools/ruby25/lib/ruby/gems/2.5.0/gems/relaton-1.3.0/lib/relaton/db.rb:156:in `check_bibliocache'
...
````

**Comment by andrew2net** [comment]
@manuel489 fixed these errors in version 1.3.3

**Comment by manuel489** [comment]
Thanks @andrew2net !
